onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -833,7 +839,100 @@ common::Status SparseTensorProtoToDenseTensorProto(const ONNX_NAMESPACE::SparseT
 
   return status;
 }
-#endif  // !defined(ORT_MINIMAL_BUILD)
+
+template <typename T>
+struct SparsifyConstants {
+  static T zero() { return 0; };
+};
+
+template <>
+struct SparsifyConstants<BFloat16> {
+  static BFloat16 zero() { return BFloat16(0.f); }
+};
+
+// Examines each of the dense elements and returns the number of elements
+// preserved. It is also a number of indicies.
+template <typename T>
+struct SparsifyDenseData {
+  Status operator()(const uint8_t* dense_raw_data, size_t n_dense_elements, TensorProto& values, TensorProto& indicies) const {
```

Comment:
Can we save a bit by doing this based on data size instead of individual types (special casing std::string)? not sure how much this method adds to binary size to say if that's worth it though.

Reply:
Actually we can exclude this from a minimal build instead. we are load only in that case so we don't need to sparsify any dense data.

---
In reply to: [510543837](https://github.com/microsoft/onnxruntime/pull/5540#discussion_r510543837) [](ancestors = 510543837)

Reply:
We sparsify in two cases. 1) Save to OrtFormat 2) When we do Model::ToProto().

---
In reply to: [510544297](https://github.com/microsoft/onnxruntime/pull/5540#discussion_r510544297) [](ancestors = 510544297,510543837)

Reply:
We don't do Model::ToProto in a minimal build (I believe it's excluded in that build type so doesn't exist)

---
In reply to: [511189229](https://github.com/microsoft/onnxruntime/pull/5540#discussion_r511189229) [](ancestors = 511189229,510544297,510543837)